### ISSUE TITLE
Problem with different host IP

### ISSUE THREAD

**Issue by danipolo** [niiknow/vestacp#20]
I'm following this project for days. I'm trying to make this work in production with no success.

That's my current docker-compose.file:

```
version: "2"   # I added version '2' to add more compatibility, my docker doesn't support it.

services:
  vesta:
    image: niiknow/vestacp:0.9.27
    container_name: vesta
    volumes:
      - "./vestacp/backup:/backup"
      - "./vestacp/vesta:/vesta"    
      - "./vestacp/home:/home"
    ports:
      - "54.39.3.6:22:22" # This is how I bind the IP. I do it with all the rest of services and works.
      - "54.39.3.6:80:80" # I simply add the IP's in /etc/network/interfaces on the host server.
      - "54.39.3.6:443:443"
      - "54.39.3.6:8083:8083"
    cap_add: ['NET_RAW', 'NET_ADMIN']
    environment:
      - "VESTA_DOMAIN=hosting.example.com"
    labels:
      - "Vesta Control Panel"
    restart: unless-stopped
```

In first instance I get this error running the docker-compose:

```
root@srv01:/home/compose/hosting(master⚡) # docker-compose up
Recreating vesta
Attaching to vesta
vesta    | *** Running /etc/my_init.d/startup.sh...
vesta    | [i] running /home/admin/bin/my-startup.sh
vesta    | bash: /home/admin/bin/my-startup.sh: No such file or directory
vesta    | [i] running /bin/vesta-auto-ssl.sh
vesta    | [err] Domain 'hosting.xscru.com' IP '54.39.3.6' does not match Host IP '37.58.28.82'
vesta    | *** Running /etc/rc.local...
vesta    | *** Booting runit daemon...
```

I went into the file and removed the IP check and it worked.

Problems starts when I try to attach new IP's to the container. How I achieve this point?

**Comment by noogen** [comment]
@danipolo per the error message, it has to do with **VESTA_DOMAIN** not matching the docker network IP address.  What you're seeing is the result of having multiple IPs.  Your docker main network is installed on the IP 37.58.28.82 and you want to use 54.39.3.6.  When my auto panel ssl script run, it fail because it didn't match.

What I can do is modify my script so you can disable this IP check.  I will make the change shortly.

====
Also, the line below in your log is weird to me
```
/home/admin/bin/my-startup.sh: No such file or directory
```
It must be a result of a bad image.  Can you recreate your local vestacp folder like so?  This will backup your existing folder and create a new one.  You can delete your old folder latest if there is nothing you want to keep.
```shell
mv ./vestacp ./vestacp-old
mkdir -p ./vestacp
```

**Comment by danipolo** [comment]
I has this now doing the mv and mkdir:

```
root@srv01:/home/compose/hosting(master⚡) # docker-compose up           
Starting vesta
Attaching to vesta
vesta    | *** Running /etc/my_init.d/startup.sh...
vesta    | ls: cannot access '/vesta/etc': No such file or directory
vesta    | [i] running /home/admin/bin/my-startup.sh
vesta    |  * Starting Apache httpd web server apache2
vesta    |  * 
vesta    |  * Starting MariaDB database server mysqld
```

**Comment by noogen** [comment]
@danipolo ok, I added the $MYIP environment variable to prevent IP check.  Clear your local folder again with something like:
```shell
rm -rf ./vestacp/*
```

Add MYIP environment variable with the value 54.39.3.6 and version 1.0.4

```yml
version: "2"   # I added version '2' to add more compatibility, my docker doesn't support it.

services:
  vesta:
    image: niiknow/vestacp:1.0.4
    container_name: vesta
    volumes:
      - "./vestacp/backup:/backup"
      - "./vestacp/vesta:/vesta"    
      - "./vestacp/home:/home"
    ports:
      - "54.39.3.6:22:22" # This is how I bind the IP. I do it with all the rest of services and works.
      - "54.39.3.6:80:80" # I simply add the IP's in /etc/network/interfaces on the host server.
      - "54.39.3.6:443:443"
      - "54.39.3.6:8083:8083"
    cap_add: ['NET_RAW', 'NET_ADMIN']
    environment:
      - "VESTA_DOMAIN=hosting.example.com"
      - "MYIP=54.39.3.6"
    labels:
      - "Vesta Control Panel"
    restart: unless-stopped
```

docker-compose up

**Comment by danipolo** [comment]
Bug here!

```
vesta    | *** Running /etc/my_init.d/startup.sh...
vesta    | [i] running for the 1st time
vesta    | [i] init nginx
vesta    | [i] init fail2ban
vesta    | [i] running /home/admin/bin/my-startup.sh
vesta    |  * Starting Apache httpd web server apache2
vesta    |  * 
vesta    |  * Starting MariaDB database server mysqld
vesta    |    ...done.
vesta    |  * Checking for corrupt, not cleanly closed and upgrade needing tables.
vesta    |  * Starting PostgreSQL 9.5 database server
vesta    |    ...done.
vesta    |  * Starting vesta-nginx vesta-nginx
vesta    |    ...done.
vesta    |  * Starting vesta-php vesta-php
vesta    |    ...done.
vesta    | [i] running /bin/vesta-auto-ssl.sh
vesta    | /bin/vesta-auto-ssl.sh: line 22: [: missing `]'
```

**Comment by noogen** [comment]
I will test my fix and let you know later tonight - it's 4:37PM CST right now.  Sorry, been a hectic day as my last day before my long vacaction.  Won't be fully available until 1/10/2018.

**Comment by danipolo** [comment]
Oh thank you! If you can solve it today will be awesome for me! I will work till late :) thank you!

**Comment by noogen** [comment]
@danipolo I fix and tested version 1.0.5, you can try it now.

**Comment by noogen** [comment]
Closing inactive issue.
